Core/include/Acts/Visualization/ViewConfig.hpp

```diff
@@ -49,17 +49,31 @@ struct Color {
   constexpr Color(double r, double g, double b)
       : Color{std::array<double, 3>{r, g, b}} {}
 
-  /// Constructor from hex string. The expected format is `#RRGGBB`
-  /// @param hex The hex string
-  constexpr explicit Color(std::string_view hex) {
-    auto hexToInt = [](std::string_view hexStr) {
-      int value = 0;
-      std::stringstream ss;
-      ss << std::hex << hexStr;
-      ss >> value;
-      return value;
+ private:
+  constexpr int hexToInt(std::string_view hex) {
+    constexpr auto hexCharToInt = [](char c) -> int {
+      if (c >= '0' && c <= '9') {
+        return c - '0';
+      } else if (c >= 'a' && c <= 'f') {
+        return c - 'a' + 10;
+      } else if (c >= 'A' && c <= 'F') {
+        return c - 'A' + 10;
+      } else {
+        throw std::invalid_argument("Invalid hex character");
+      }
     };
 
+    int value = 0;
+    for (char c : hex) {
+      value = (value << 4) + hexCharToInt(c);
+    }
+    return value;
+  };
+
+ public:
+  /// Constructor from hex string. The expected format is `#RRGGBB`
+  /// @param hex The hex string
+  constexpr Color(std::string_view hex) {
```

Comment:
I intentionally made this non `explicit`, because it makes code in GCC more verbose, but nor sonar complains about this.